crypto/pkcs7/pkcs7.c

```diff
@@ -461,6 +457,27 @@ ASN1_TYPE *PKCS7_get_signed_attribute(const PKCS7_SIGNER_INFO *si, int nid) {
   return NULL;
 }
 
+ASN1_TYPE *PKCS7_get_signed_attribute(const PKCS7_SIGNER_INFO *si, int nid) {
+  if (si == NULL) {
+    OPENSSL_PUT_ERROR(PKCS7, ERR_R_PASSED_NULL_PARAMETER);
+    return NULL;
+  }
+  return get_attribute(si->auth_attr, nid);
+}
+
+static ASN1_OCTET_STRING *PKCS7_digest_from_attributes(
+    STACK_OF(X509_ATTRIBUTE) *sk) {
+  if (sk == NULL) {
```

Comment:
nit: should we use `GUARD_PTR`?

Reply:
It works and we've contemplated it before in similar instances, but there were concerns with `GUARD_PTR` returning a `0` and returning `0` as `NULL` being un-guaranteed behavior within C
* https://c-faq.com/null/runtime0.html